### ISSUE TITLE
Simplify receiver registration process

### ISSUE THREAD

**Issue by j-sayers** [MotusDev/Motus-TO-DO#1120]
The current options presented for receiver registration are quite confusing (below). 

![image](https://user-images.githubusercontent.com/55001144/157064736-51bb2fcc-f4c6-4e40-9158-98473ffd9ccc.png)

I think if we split this up it would make the process easier to navigate

1) First option only presents the 4 supported receiver types

![image](https://user-images.githubusercontent.com/55001144/157065104-c4c548ac-fefb-45ba-a8eb-46993615e7c9.png)

2) After a receiver type has been selected, the naming convention for that receiver type can be given. For Lotek receivers this allow allows a better opportunity to [confirm whether the receiver is an SRX 1200](#1113).

**Comment by leberrigan** [comment]
Yeah this really needs some work, eh? I think your idea makes a lot more sense and is not so overwhelming to deal with. The problem here is that we're beginning to pile up a list of these improvements. Perhaps we should keep an updated list somewhere, like here? https://github.com/MotusDev/Motus-TO-DO/wiki/Fixes-to-Beta-10

**Comment by leberrigan** [comment]
@j-sayers Can you identify the discrete steps required for this and assign the next one?

**Comment by leberrigan** [comment]
That sounds like a simple conditional form - I can write a bit of JS to make that happen, but I'm a bit worried about doing code insertions myself in case I break something. Otherwise it'll probably have to be @Hussell when he has the time.

**Comment by j-sayers** [comment]
Also see #1113. It's unclear if Lotek SRX model matters unless it's a SRX 1200, other than to match processed data with registered receivers...

**Comment by cjardine-bsc** [comment]
https://sandbox.motus.org/data/project/receivers?projectID=1

Click on "Add Receiver"

**Comment by bcreightonyoung** [comment]
I'm working on this one right now and it's looking a LOT better! I just wanted to clarify a few things:

1. I should definitely remove the "unknown" option from this form?
2. SEI refers to Sigma Eight, sensorstation is CTT, correct?

**Comment by j-sayers** [comment]
Awesome!

Yes that’s correct.

On Mar 8, 2023, at 11:10, bcreightonyoung ***@***.***> wrote:

﻿

I'm working on this one right now and it's looking a LOT better! I just wanted to clarify a few things:

  1.  I should definitely remove the "unknown" option from this form?
  2.  SEI refers to Sigma Eight, sensorstation is CTT, correct?

**Comment by bcreightonyoung** [comment]
@j-sayers this is on sandbox I think it is a little bit more functional now. I am fairly confident that it is fully-functional, i.e. I didn't break things. But is it always a possibility.

I also added a correct text fragment for the SRX 1200 receiver serial number (based on the regu;ar expressions I found in the database). Feedback welcome.

**Comment by j-sayers** [comment]
Great! Here are a couple suggestions:

- For SensorGnome, SensorStation, and Sigma Eight, remove the secondary radio buttons and replace with the appropriate description of the naming format
- Keep the addition additional radio buttons and add the relevant formats. Sure is a mess with all though, though, and I'm not sure the Lotek model is critical to know unless it's a SRX 1200 due to the time zone....

**Comment by bcreightonyoung** [comment]
Good idea for the first bullet. Forms are finicky but I think I can get that going. For the second though, wouldn't it be wise to keep the particular Lotek model for data integrity purposes? It provides a decent check to ensure the IDs are entered properly, too.

**Comment by j-sayers** [comment]
>  For the second though, wouldn't it be wise to keep the particular Lotek model for data integrity purposes? It provides a decent check to ensure the IDs are entered properly, too

Yeah I was mostly just musing but I agree, it's best to keep the whole gang :) At any rate, I don't even know that the model isn't needed for anything else so best to keep it all for now!

**Comment by bcreightonyoung** [comment]
@j-sayers the new-and-improved version should be up on sandbox, let me know what you think.

As a side note, the "expected ID format" text fragments that are displayed in the form are not actually correct. I spent a while convinced that I broke the form while testing it, as each time I tried to add a test receiver I got thrown a "does not match the expected pattern" warning. It wasn't until I looked at the regular expressions in the database that I realize that the formats displayed to the user are under-specific. 

Of course, this doesn't matter _too_ much to a user who has a receiver with a valid serial number in their hands. Also it could limit fraudulent/non-existent/incorrect receiver IDs from being added to the database. I was just wondering if you were aware of this and what you think of it.
